src/utils/MSFraggerAdapter.cpp

```diff
@@ -0,0 +1,865 @@
+          << "\noverride_charge = " << (arg_override_charge ? 1  : 0)
+          << "\n\ndigest_min_length = " << arg_digest_min_length
+          << "\ndigest_max_length = " << arg_digest_max_length
+          << "\ndigest_mass_range = " << arg_digest_mass_range_min << " " << arg_digest_mass_range_max
+          << "\nmax_fragment_charge = " << arg_max_fragment_charge
+          << "\n\ntrack_zero_topN = " << arg_track_zero_topn
+          << "\nzero_bin_accept_expect = " << arg_zero_bin_accept_expect
+          << "\nzero_bin_mult_expect = " << arg_zero_bin_mult_expect
+          << "\nadd_topN_complementary = " << arg_add_topn_complementary
+          << "\n\nminimum_peaks = " << arg_minimum_peaks
+          << "\nuse_topN_peaks = " << arg_use_topn_peaks
+          << "\nmin_fragments_modelling = " << arg_min_fragments_modeling
+          << "\nmin_matched_fragments = " << arg_min_matched_fragments
+          << "\nminimum_ratio = " << arg_minimum_ratio
+          << "\nclear_mz_range = " << arg_clear_mz_range_min << " " << arg_clear_mz_range_max
+          << "\nadd_Cterm_peptide = " << arg_add_cterm_peptide
+          << "\nadd_Nterm_peptide = " << arg_add_nterm_peptide
+          << "\nadd_Cterm_protein = " << arg_add_cterm_protein
+          << "\nadd_Nterm_protein = " << arg_add_nterm_protein
+          << "\n\nadd_G_glycine = " << arg_add_G_glycine
+          << "\nadd_A_alanine = " << arg_add_A_alanine
+          << "\nadd_S_serine = " << arg_add_S_serine
+          << "\nadd_P_proline = " << arg_add_P_proline
+          << "\nadd_V_valine = " << arg_add_V_valine
+          << "\nadd_T_threonine = " << arg_add_T_threonine
+          << "\nadd_C_cysteine = " << arg_add_C_cysteine
+          << "\nadd_L_leucine = " << arg_add_L_leucine
+          << "\nadd_I_isoleucine = " << arg_add_I_isoleucine
+          << "\nadd_N_asparagine = " << arg_add_N_asparagine
+          << "\nadd_D_aspartic_acid = " << arg_add_D_aspartic_acid
+          << "\nadd_Q_glutamine = " << arg_add_Q_glutamine
+          << "\nadd_K_lysine = " << arg_add_K_lysine
+          << "\nadd_E_glutamic_acid = " << arg_add_E_glutamic_acid
+          << "\nadd_M_methionine = " << arg_add_M_methionine
+          << "\nadd_H_histidine = " << arg_add_H_histidine
+          << "\nadd_F_phenylalanine = " << arg_add_F_phenylalanine
+          << "\nadd_R_arginine = " << arg_add_R_arginine
+          << "\nadd_Y_tyrosine = " << arg_add_Y_tyrosine
+          << "\nadd_W_tryptophan = " << arg_add_W_tryptophan;
+      os.close();
+    }
+    catch (int)
+    {
+      return ILLEGAL_PARAMETERS;
+    }
+
+    QStringList process_params; // the actual process is Java, not MSFragger
+    process_params << "-Xmx" + QString::number(this->getIntOption_(TOPPMSFraggerAdapter::param_java_heapmemory)) + "m"
+        << "-jar" << this->executable.toQString()
+        << this->parameter_file_path.toQString();
+
+    for (const QString & input_file : this->input_files)
```

Comment:
can you add a debug level output of the command line call? This may allow people to reconstruct what was done and potentially what went wrong?

Reply:
Done